### PR TITLE
EPL-8413 feat(package): update draft4 tests

### PR DESCRIPTION
**Pull request by michaelcarter-wf**
## Ultimate problem:
- Repo was using an old copy of the [JSON Schema Test Suite](https://github.com/json-schema-org/JSON-Schema-Test-Suite).
- Updated test cases exposed some bad behavior.

## How it was fixed:
- Update the test suite
- Fix the bad behaviors

## Testing suggestions:
- ddev test

## Potential areas of regression:
- This could be breaking for consumers who were dependent on the bad behavior.

---

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Module Findings
No security relevant content detected. Please review for security relevance and request security review as needed.

**Comment by semveraudit-wf** [comment]
### Public API Changes

 No changes to the public API found for commit e3ec36eabaa17c76501e09f0d16776aad61abcf2

Showing results for e3ec36eabaa17c76501e09f0d16776aad61abcf2
> Powered by [semver-audit-service](https://github.com/Workiva/semver-audit-service). Please report any problems by [filing an issue](https://github.com/Workiva/semver-audit-service/issues).
[Browse public API](https://semver-audit.wk-dev.wdesk.org/Workiva/json_schema/e3ec36eabaa17c76501e09f0d16776aad61abcf2).

**Comment by jamesromo-wf** [comment]
@michaelcarter-wf CI is looking good minus an issue with the dependency validator
```
The command "pub run dependency_validator --ignore dart_style,coverage" exited with 1.
```

File: test/unit/vm/json_schema/validation_test.dart [hunk 1]
```diff
@@ -97,7 +97,15 @@ void main([List<String> args]) {
 
           validationTests.forEach((validationTest) {
             final String validationDescription = validationTest['description'];
-            test('${description} : ${validationDescription}', () {
+            final String testName = '${description} : ${validationDescription}';
+
+            // Individual test cases to skip - reason listed in comments
```
Comment (jamesromo-wf) [inline test/unit/vm/json_schema/validation_test.dart]: #nit `.` at the end to match other cases?

**Comment by jamesromo-wf** [comment]
+10 looks good!

**Pull request merged**